src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs

```diff
@@ -234,5 +246,8 @@ private string GetMimeType()
 
 			return mimeTypes.ToArray();
 		}
+
+		internal void RegisterOnBeforeStartActivity(Intent intent)
```

Comment:
I think it would be more practical if this took an `Action<Intent>` which would execute when the dialog is being opened, as currently the second `Intent` instance is created only to carry the flags. With an action the user could just use the internal `Intent` directly and potentially modify more than just the flags.

Reply:
You are right this makes elegant use possible and opens up more possibilities. I have implemented this.